### ISSUE TITLE
[rllib] dqn torch policy incorrect support probabilities for num_atoms>1

### ISSUE THREAD

**Issue by ssykiotis** [ray-project/ray#15137]
In rllib's DQN Torch policy implementation, I had an issue where all the q-values were set to the same value. After some digging and debugging, I figured out that the issue lies in the softmax function in dqn_torch_policy.py, line 360. In the nn.functional.softmax() call, dim=-1 is required for the code to run properly if num_atoms>1. Otherwise all probabilities are set to 1.

Software versions:
Ray 1.2.0 ,Python 3.7.10, Pytorch 1.7.0 , Ubuntu 20.04/Google colab)
